### PR TITLE
Set defaults to query ingesters, not store, for recent data

### LINKED ISSUE #1639: Change default -querier.query-store-after to 12h
We recommend to query last 12h from ingesters only but default config still has `-querier.query-store-after=0s`. We should change `-querier.query-store-after` default to 12h and ensure the rest of the config is consistent with that.

An option of new defaults:

- `-querier.query-store-after=12h`
- `-blocks-storage.bucket-store.ignore-blocks-within=10h`

### PR DESCRIPTION
**Pull request by 56quarters**
#### What this PR does

Set queriers to _not_ query storage (store-gateways) for recent data
and set the store-gateways to ignore recent uncompacted blocks.

Default values are set to match what we use in the Mimir jsonnet.

#### Which issue(s) this PR fixes or relates to

Fixes #1639

Signed-off-by: Nick Pillitteri <nick.pillitteri@grafana.com>

#### Checklist

- [X] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by 56quarters** [comment]
I will remove these configuration settings from our Jsonnet + helm chart in follow up PRs.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 ### Grafana Mimir
 
 * [CHANGE] Increased default configuration for `-server.grpc-max-recv-msg-size-bytes` and `-server.grpc-max-send-msg-size-bytes` from 4MB to 100MB. #1883
+* [CHANGE] Default values have changed for the following settings: #1909
```
Comment (pracucci) [inline CHANGELOG.md:8]: I would slightly explain the explanation saying that we're doing this change to improve query performance using the default config. The trade-off is that the last 12h of data is always queried only from ingesters and never from store-gateways.

**Review by pracucci — APPROVED** [review]
Thanks for working on this! LGTM (I just left a comment about the CHANGELOG entry).

For the record, I've also checked `-querier.query-ingesters-within` which defaults to 13h, so it's good.

I'm ok if you want to do jsonnet changes in a follow up PR, but please remember about it ;)

File: docs/sources/operators-guide/running-production-environment/production-tips/index.md [hunk 2]
```diff
@@ -46,14 +46,15 @@ When running Grafana Mimir at scale, querying non-compacted blocks might be inef
...
+
+Configure Grafana Mimir so large tenants are parallelized by the compactor:
 
 1. Configure compactor's `-compactor.split-and-merge-shards` and `-compactor.split-groups` for every tenant with more than 20 million active series. For more information about configuring the compactor's split and merge shards, refer to [compactor]({{< relref "../../architecture/components/compactor/index.md" >}}).
-1. Configure querier's `-querier.query-store-after` equal to `-querier.query-ingesters-within` minus five minutes. The five-minute delta is recommended to ensure the time range on the boundary is queried both from ingesters and queriers.
 
 #### How to estimate `-querier.query-store-after`
 
-Set the `-querier.query-store-after` to a duration that is large enough to give compactor enough time to compact newly uploaded blocks, and queriers and store-gateways to discover and synchronize newly compacted blocks.
+If not using the defaults, set the `-querier.query-store-after` to a duration that is large enough to give compactor enough time to compact newly uploaded blocks, and queriers and store-gateways to discover and synchronize newly compacted blocks.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/running-production-environment/production-tips/index.md:57]: +you are @pracucci

Comment (56quarters) [reply docs/sources/operators-guide/running-production-environment/production-tips/index.md:57]: Incorporated into a new followup PR: https://github.com/grafana/mimir/pull/1915

**Pull request merged**